test/WebSocketSessionTest.cpp

```diff
@@ -157,9 +178,68 @@ ASYNC_TEST(WebSocketSession, verifySessionEndsWhenServerIsDoneSending) {
     boost::beast::flat_buffer buffer;
     http::request<http::string_body> request;
     co_await http::async_read(stream, buffer, request, net::use_awaitable);
-    co_await WebSocketSession::handleSession(queryHub, request,
+    co_await WebSocketSession::handleSession(queryHub, registry, request,
+                                             std::move(stream.socket()));
+  };
+
+  co_await net::co_spawn(strand, serverLogic() && controllerActions(),
+                         net::use_awaitable);
+}
+
+// _____________________________________________________________________________
+
+ASYNC_TEST(WebSocketSession, verifyCancelStringTriggersCancellation) {
+  auto strand = net::make_strand(ioContext);
+  QueryHub queryHub{ioContext};
+  QueryRegistry registry;
+
+  auto queryId = registry.uniqueIdFromString("some-id");
+  auto cancellationHandle =
+      registry.getCancellationHandle(queryId->toQueryId());
+
+  tcp::socket server{strand};
+  tcp::socket client{strand};
+  co_await connect(server, client);
```

Comment:
A lot of this code is redundant boilerplate that happens in many of these test cases. 
Can you consider setting up the `client/server/strand` pair in a helper function / fixture / etc.